threat-model/readme.md

```diff
@@ -230,11 +204,53 @@ When managing a privacy budget, that budget is assigned to some party who can us
+
+In terms of our threat model, these parties are aggregators and we assume that an attacker can control some subset of those aggregators. That exact threshold may be different for a given proposal, for example, we may assume that an attacker can only control one out of three aggregators. This would enable, in cryptographic terms, a maliciously secure, honest two out of three majority MPC.
+
+Given these aggregators, a _client/user_ is able to generate a secret sharing of their input data, and then securely communicate one secret share to each aggregator. This allows the aggregators to then perform the MPC protocol to compute the predefined function. The _client/user_ is assuming that an attacker cannot control enough aggregators to violate that protocol, which implies that they only perform the expected function.
+
+### 4.2 Trusted Execution Environments (TEEs)
+
+Trusted execution environments are specialized hardware where encrypted data can be sent "in" to an enclave where it is decrypted and operated on, but cannot be otherwise accessed. It also produces an "attestation" that provides a means of verifying that the code run on that data is the code that was intended. Different hardware manufacturers offer different types of TEEs, and there are various assumptions which need to be made about the manufacturer, hardware operator, and other tenants on the hardware in order to achieve our ideal of input data remaining oblivious. There are documented attacks for many of these types of hardware, and while we don't expect all web platform vendors to support this construction, some vendors have expressed comfort with the required assumptions.
+
+In order to utilize a TEE to achieve private computation, we need two properties. First, the data must only be able to be decrypted within the TEE, and second, we need to assure that the TEE only executes code which evaluates the predefined function.
+
+In a very simplified form, we could imagine there is one TEE which is used by every _first party_/_delegated party_ to run the predefined function. In this case, the _user/client_ could encrypt their data with the public key of the TEE. However, this ignores assuring the code running is that that's intended. The TEE produces an attestation of the code it's running, however the _client/user_ needs that attestation to be verified before their input data is decrypted.
+
+This can be achieved by utilizing an aggregator who verifies the TEE attestation on behalf of the _client/user_. In this case, the _client/user_ encrypts the data towards the aggregator, and the aggregator provides the decryption key into the TEE if and only if the attestation verifies the TEE is only running the expected code. This still has one problem, however, as we assume that our attacker can compromise at least one aggregator.
```

Comment:
Michael is using "coordinator" here rather than "aggregator".  The TEE is an aggregator here as well.

Reply:
Yes, I was hoping that we might be able to use the same term, since the threats are similar, and it allows for the nice abstraction of "browser just support different aggregator networks." 

Another approach would be to separately include "coordinators" as types of actors above, and then have a meta class of actors, something like "helper parties", of which both "coordinators" and "aggregators" are. 

> The TEE is an aggregator here as well.

The TEE itself isn't an actor, right? We do have the cloud provider (which upon reflection, we should add in general, as it has implications for both) and the TEE manufacturer (which may also need to be added.)

Reply:
I think that you are right in attempting to identify the entities that have a critical role in protecting the data.  For TEEs that is both the vendor and the operator.  I guess for an MPC, each node has to contend with operational and procedural issues, but we are generally comfortable lumping those together under a single umbrella as the same entity is on the hook for both, even if they subcontract aspects of the operation out.

Reply:
> but we are generally comfortable lumping those together under a single umbrella as the same entity is on the hook for both, even if they subcontract aspects of the operation out.

Yes, but I think we aren't comfortable with two distinct entities running MPC within a single cloud provider.

Reply:
Right.  Though some might be, not all of us are.  Still, it falls under the rubric of having some form of validation of the MPC operations as a whole.